crates/sled/src/ivec.rs

```diff
@@ -125,6 +125,21 @@ impl From<Vec<u8>> for IVec {
     }
 }
 
+macro_rules! from_array {
+    ($($s:expr)*) => {
+        $(
+            impl From<&[u8; $s]> for IVec {
+                fn from(v: &[u8; $s]) -> IVec {
+                    IVec::from(&v[..])
+                }
+            }
+        )*
+    }
+}
+
+from_array!(0 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15);
+from_array!(16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32);
```

Comment:
Why splitting these macro calls? https://doc.rust-lang.org/src/core/array.rs.html#258-263
I know, I am a little picky 😄

Reply:
I'd prefer keeping these together as well but we can fix that quickly later

Reply:
I just realized that it is probably not the right implementation needed, we would probably have prefered `impl From<[u8; $n]> for IVec {}` without the ampersand.

Reply:
Can you explain why `From<[u8; $n]>` is better? I was mainly just going according to the existing `From` implementations: there's `From<&str>` and `From<&[u8]>`.

Reply:
I was thinking about [the `Vec` `PartialEq` impls](https://doc.rust-lang.org/std/vec/struct.Vec.html#impl-PartialEq%3C%5BB%3B%205%5D%3E) that allows to compare arrays with a `Vec` without the need to make the array be a slice (`&array[]..`).

An array of `u8` implements `Copy` anyway so it is a little bit disturbing to force the user to put an ampersand in front of its array, making the construction be `IVec::from(&[3u8; 0])` instead of the simplest `IVec::from([3u8; 0])`.

Reply:
Well, `Vec` does it for both `&[u8; $n]` and `[u8; $n]`, so maybe we should also? The owned version is more convenient if your argument is `[3u8; 1]`, but the borrowed one is more convenient if your argument is `b"yo"`.

Reply:
You are totally right! We should probably implement `PartialEq`/`Eq` for `IVec` and `[u8; $n]`/`&[u8]` but should we do the same for `From<[u8; $n]>`? Wouldn't it be better to follow the std `Vec` implementations?

Reply:
I'm not sure I follow. The std `Vec` doesn't implement `From` for any arrays, borrowed or otherwise. What do you mean by "follow the std `Vec` implementations"?

Reply:
I mean implementing `PartialEq`, sorry for the confusion :)

Reply:
So I looked into `PartialEq`, and I don't think any more implementations are necessary, because we have `PartialEq<T>` whenever `T: AsRef<[u8]>`, which covers both borrowed and owned arrays.

Unfortunately, this route doesn't seem to be possible for `From`, because of coherence: `IVec` implements `AsRef<[u8]>`, so `impl<T: AsRef<[u8]>> From<T> for IVec` conflicts with the blanket `From<T> for T` implementation.

Reply:
I wonder why `Vec<T> where T: u8` do not implement `PartialEq<U> where U: AsRef<[u8]>`? 
Is it related to specialization?